### DISCUSSION TITLE
Please share your BBS-FW feedback here

### DISCUSSION THREAD

**Discussion by silenec** [danielnilsson9/bbs-fw#84] [General]
I would like to flash this FW but I feel like there is almost no feedback from the users so I am not really sure if it works for everyone or not.
If you have a minute, please consider filling out this "form" below to share your success or failure story. Thanks!

-------------

Hardware: BBSHD / BBS02B / BBS02A / TSDZ2 / Other (specify)

MCU type: STC15W4K56S4 / IAP15W4K61S4 / STC15F2K60S2 / IAP15F2K61S2 / ???

Year of manufacture or since when do you own the motor if not available:

Other details about your hardware (stock FW version, sticker numbers etc.):

Did you brick your controller while flashing BBS-FW? Yes / No

Have you tried flashing back to stock FW for any reason? Yes / No
If yes, did it work and what .hex/.bin filename did you use?

What works or does not work for you with BBS-FW and your other notes:

**Comment by zeynelozturk** [comment]
BBSHD 

MCU: IAP15W4K61S4 (not 100% sure, but I remember I have the new one)

Motor age: About 1,5 years. Bought in July 2021.

Other details: Prior to BBS-FW I had installed the "known" "52V" firmware available on endless-sphere  and it was fine. Because the controller sometimes wouldn't like and give error nearing 58.8V fully charged.

Going back: No.

BBS-FW is fine except I need to be careful about shift sensor cutoff under >250W load. It's probably too short for now. I also needed to switch "9 pas levels" on the screen as I couldn't map 5 levels to the software, but that could be my fault. Anyway it doesn't matter because you can skip, or use whichever PAS level you want on the new config tool.

**Comment by Gonespa** [comment]
Hardware: BBS02B

MCU type: STC15F2K60S2

Year of manufacture or since when do you own the motor if not available: at least since 2019

Other details about your hardware (stock FW version, sticker numbers etc.): stock FW version 7.2.5S

Did you brick your controller while flashing BBS-FW? No

Have you tried flashing back to stock FW for any reason? No
If yes, did it work and what .hex/.bin filename did you use?

What works or does not work for you with BBS-FW and your other notes:

For me, there is 1 issue that shows in this firmware for my very specific use case scenario: I usually use it like some kind of light motorbike. So I configured maximum amperage (30 or even 33 Amps). I noticed 2 or 3 times that the motor cut for a second in high load scenarios. Apparently the NEC IC detects some situation in that case and sends an error signal. This does not show in original fw as that one does not let you go over 25 amps.
Other than that, my display (500c) also apparently does not correctly show Watt values over thise corresponding to 26 Amps. Again not an issue of this fw.

Other than that, with this fw you gave more torque available, yo have the posibility to show HW temp on the fly, you can finally pedal and use throttle at the same time, it also allows for a very quick engangement when starting pedaling ( it anoyed me A LOT on the legacy fw, since that have a huge ramp up of several seconds to the configured current of the pas level).

And last but not least, with the original fw my battery was expected to be 52v. Even if I changed that in the display to 48 it didnt change anything, meaning I had like only half the bar available losing a lot of precision. With this you can define max voltage to the first decimal.

**Comment by DesQu** [comment]
Hardware: BBSHD

MCU type: IAP15W4K61S4 if I remember correctly.

Year of manufacture or since when do you own the motor if not available: Bought early 2023

Other details about your hardware (stock FW version, sticker numbers etc.):

Did you brick your controller while flashing BBS-FW? No

Have you tried flashing back to stock FW for any reason? No
If yes, did it work and what .hex/.bin filename did you use?

What works or does not work for you with BBS-FW and your other notes:
The manufacturer firmware was 1.4, I found that using the throttle was jumpy and seemed to impact the proper working of the cadence operation, I tried changing the settings, lowering the maximum speed etc, while that improved matters the throttle was still jumpy and it was unpleasant to ride. 
Installing the BBS-FW is a revelation, the bike is now smooth and a delight to ride.

**Reply by maddog75** [reply]
Same for me. BBS-FW on my BBSHD has made everything better! Smoother ramp up of power at takeoff, throttle works perfectly giving a boost up to 100% irrespective of the current PAS current limit. I should have done the firmware switch sooner!
